web/concrete/js/build/core/translator.js

```diff
@@ -0,0 +1,720 @@
+/* jshint unused:vars, undef:true, browser:true, jquery:true */
+(function() {
+'use strict';
+
+if(window.ccmTranslator) {
+  return;
+}
+
+var MAX_TRANSLATIONS_FOR_FASTSEARCH = 500;
+
+var i18n = {
+  AskDiscardDirtyTranslation: 'The current item has changed.\nIf you proceed you will lose your changes.\n\nDo you want to proceed anyway?',
+  Comments: 'Comments',
+  Context: 'Context',
+  ExamplePH: 'Example: %s',
+  Filter: 'Filter',
+  Original_String: 'Original String',
+  Please_fill_in_all_plurals: 'Please fill-in all plural forms',
+  Plural_Original_String: 'Plural Original String',
+  References: 'References',
+  Save_and_Continue: 'Save & Continue',
+  Search_for_: 'Search for...',
+  Search_in_contexts: 'Search in contexts',
+  Search_in_originals: 'Search in originals',
+  Search_in_translations: 'Search in translations',
+  Show_approved: 'Show approved',
+  Show_translated: 'Show translated',
+  Show_unapproved: 'Show unapproved',
+  Show_untranslated: 'Show untranslated',
+  Singular_Original_String: 'Singular Original String',
+  Toggle_Dropdown: 'Toggle Dropdown',
+  TAB: '[TAB] Forward',
+  TAB_SHIFT: '[SHIFT]+[TAB] Backward',
+  Translate: 'Translate',
+  Translation: 'Translation',
+  PluralNames: {
+    zero: 'Zero',
+    one: 'One',
+    two: 'Two',
+    few: 'Few',
+    many: 'Many',
+    other: 'Other'
+  }
+};
+
+var translatorsCount = 0;
+
+function Translation(data, translator) {
+  $.extend(this, data);
+  this.hasContext = 'context' in data;
+  this.isPlural = 'originalPlural' in data;
+  this.isTranslated = 'translations' in data;
+  this.translator = translator;
+  this.translator.translations.push(this);
+}
+Translation.prototype = {
+  buildListItem: function() {
```

Comment:
Here we have some bad coding lines, but this code have to be as fast as possible